### ISSUE TITLE
Suggestion: Pathfinding, flocking behaviours?

### ISSUE THREAD

**Issue by JohnyDL** [TangoTek/TekTopia-Community#53]
Villagers seem to like rubbing their faces against walls walking over things that're against walls and so on, maybe adding some villager flocking behaviours could make them feel more natural?

Avoid walls: try to stay 2+ blocks from any blocks at head height if possible when pathing
Avoid crowding: don't enter a 3x3x3 area with more than 4 villagers in it if possible
Safety in numbers: when not able to do anything else (say night time or no job to do) move 10 steps towards the average location of 5 random (not guard) villagers in the village

I'm sure there are other flocking type behaviours that could be added to make villagers feel more natural and alive

**Comment by MilesKid** [comment]
I noticed that if you build a structure after Villagers have established paths they will try and go through the block and get stuck at the block.  Nudging them away helps.  (Server 2 - MilesVille)

**Comment by JohnyDL** [comment]
yeah my building was prebuilt on creative months ago but I put lighting in the corners and they decide to climb chests and bump their heads on the end rods and get jammed. in reality, people would take a less efficient route to avoid the confinement feeling of being next to a wall
